### PR TITLE
fix: enhance the error message when agent change customer's password

### PR DESCRIPTION
**Pull request by i53577**
JIRA: https://jira.tools.sap/browse/CXSPA-3399

### CODE REVIEW

File: feature-libs/user/profile/components/update-password/update-password-component.service.ts [hunk 1]
```diff
@@ -93,7 +94,12 @@ export class UpdatePasswordComponentService {
     });
   }
 
-  protected onError(_error: Error): void {
+  protected onError(_error: HttpErrorModel): void {
```
Comment (scarai-sap) [inline feature-libs/user/profile/components/update-password/update-password-component.service.ts]: changed the signature of the routine?

File: feature-libs/user/profile/components/update-password/update-password-component.service.ts [hunk 2]
```diff
@@ -93,7 +94,12 @@ export class UpdatePasswordComponentService {
...
+    if (_error.details?.[0].type === 'AccessDeniedError')
+    this.globalMessageService.add(
```
Comment (scarai-sap) [inline feature-libs/user/profile/components/update-password/update-password-component.service.ts]: code not aligned, and always use {} even we only have one line inside the if

Comment (evy-yang01) [reply feature-libs/user/profile/components/update-password/update-password-component.service.ts]: strange, it worked as expected. Indeed I missed the {}, I will add it. Nice observation

**Review by scarai-sap — DISMISSED** [review]

**Review by scarai-sap — DISMISSED** [review]

File: feature-libs/user/profile/components/update-password/update-password-component.service.ts [hunk 3]
```diff
@@ -93,7 +94,16 @@ export class UpdatePasswordComponentService {
     });
   }
 
-  protected onError(_error: Error): void {
+  protected onError(_error: HttpErrorModel | Error): void {
```
Comment (kimhw0630) [inline feature-libs/user/profile/components/update-password/update-password-component.service.ts]: I spoke with Patrick Labadie (@plabadie ), and he mentioned that this change could potentially lead to compilation errors in the customer's custom code.  so we might need to wait for a major version

Comment (scarai-sap) [reply feature-libs/user/profile/components/update-password/update-password-component.service.ts]: if customer call onError with type Error, it should pass the compilation. Why it make potential compilation error?

**Pull request closed without merging**